stream_alert/alert_processor/outputs.py

```diff
@@ -1054,3 +1056,285 @@ def dispatch(self, **kwargs):
+        field of an issue.
+
+        Args:
+            issue_name (str): The name of the Jira issue
+            project_key (str): The Jira project key which issues will be associated with
+            issue_type (str): The type of issue being created
+            description (str): The body of text which describes the issue
+
+        Returns:
+            int: ID of the created issue or False if unsuccessful
+        """
+        issue_url = os.path.join(self._base_url, self.ISSUE_ENDPOINT)
+        issue_body = {
+            'fields': {
+                'project': {
+                    'key': project_key
+                },
+                'summary': issue_name,
+                'description': description,
+                'issuetype': {
+                    'name': issue_type
+                }
+            }
+        }
+
+        resp = self._post_request(issue_url,
+                                  data=issue_body,
+                                  headers=self._get_headers(),
+                                  verify=False)
+
+        success = self._check_http_response(resp)
+        if not success:
+            return False
+
+        return resp.json()['id']
+
+    def _establish_session(self, username, password):
+        """Establish a cookie based Jira session via basic user auth.
+
+        Args:
+            username (str): The Jira username used for establishing the session
+            password (str): The Jira password used for establishing the session
+
+        Returns:
+            str: Header value intended to be passed with every subsequent Jira request
+                 or False if unsuccessful
+        """
+        login_url = os.path.join(self._base_url, self.LOGIN_ENDPOINT)
+        auth_info = {'username': username, 'password': password}
+
+        resp = self._post_request(login_url,
+                                  data=auth_info,
+                                  headers=self._get_default_headers(),
+                                  verify=False)
+
+        success = self._check_http_response(resp)
+        if not success:
+            LOGGER.error("Failed to authenticate to Jira")
+            return False
+        resp_dict = resp.json()
+
+        return '{}={}'.format(resp_dict['session']['name'],
+                              resp_dict['session']['value'])
+
+    def dispatch(self, **kwargs):
+        """Send alert to Jira
+
+        Args:
+            **kwargs: consists of any combination of the following items:
+                descriptor (str): Service descriptor (ie: slack channel, pd integration)
+                rule_name (str): Name of the triggered rule
+                alert (dict): Alert relevant to the triggered rule
+        """
+        creds = self._load_creds(kwargs['descriptor'])
+        if not creds:
+            return self._log_status(False)
+
+        issue_id = None
+        comment_id = None
+        issue_summary = 'StreamAlert {}'.format(kwargs['rule_name'])
```

Comment:
We should consider moving strings like this into class definitions to easily change/modify in the future cc @ryandeivert